scripts/mod_loader/modapi/palette.lua

```diff
@@ -0,0 +1,354 @@
+
+local canAddPalettes = true
+local defaultPaletteOrder
+
+local palettes = {
+	index = {},
+	hash = {},
+	palette = {},
+	redirect = {},
+	image = {},
+	
+	getIndex = function(self, hash)
+		return self.index[hash]
+	end,
+	
+	getHash = function(self, index)
+		return self.hash[index]
+	end,
+	
+	getPalette = function(self, index)
+		local hash = self.hash[index]
+		if not hash then return nil end
+		
+		return self.palette[hash]
+	end,
+	
+	getCount = function(self)
+		return #self.hash
```

Comment:
My version of this function looked like:
```lua
    local count = #self.hash
    if count > 11 and sdlext.isHangar() then
      return 11
    end
    return count
```

Effectively it caused only 11 palettes to show in the hangar instead of showing extra ones with broken buttons, similar to the pilot patch

Reply:
I'm not sure I understand what you mean by "extra ones with broken buttons"